src/gadgets/record.rs

```diff
@@ -1,16 +1,21 @@
-use super::UInt64Gadget;
+use super::{UInt64Gadget, AddressGadget};
 
 #[derive(Clone, Debug)]
 pub struct Record {
+    address: AddressGadget,
     gates: UInt64Gadget,
 }
 
 impl Record {
-    pub fn new(gates: UInt64Gadget) -> Self {
-        Self { gates }
+    pub fn new(address: AddressGadget, gates: UInt64Gadget) -> Self {
+        Self { address, gates }
```

Comment:
I think that this `new` method is not necessary, the user can create a new element just using the struct